src/Api/Rest/Rest.php

```diff
@@ -0,0 +1,225 @@
+<?php
+/**
+ * @package     Redcore
+ * @subpackage  Api
+ *
+ * @copyright   Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Webservices\Api\Rest;
+
+use Joomla\Webservices\Api\ApiBase;
+use Joomla\Webservices\Resource\Item;
+use Joomla\Webservices\Resource\Resource;
+use Joomla\Webservices\Webservices\Webservice;
+use Joomla\Webservices\Webservices\Factory;
+
+use Joomla\DI\Container;
+use Joomla\Event\Event;
+use Joomla\Event\EventImmutable;
+use Joomla\Registry\Registry;
+
+/**
+ * Class to represent a REST interaction style.
+ *
+ * @since  1.2
+ */
+class Rest extends ApiBase
+{
+	/**
+	 * @var    string  Name of the Api
+	 * @since  1.2
+	 */
+	public $apiName = 'rest';
+
+	/**
+	 * Profile object.
+	 * 
+	 * @var  Profile
+	 */
+	private $profile = null;
+
+	/**
+	 * Main resource object
+	 * @var  Resource
+	 */
+	public $resource = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param   Container  $container  The Dependency Injection Container object.
+	 * @param   Registry   $options    Optional custom options to load.
+	 *
+	 * @throws  \Exception
+	 * 
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function __construct(Container $container, Registry $options)
+	{
+		parent::__construct($container, $options);
+
+		// Set initial status code.
+		$this->setStatusCode(200);
+	}
+
+	/**
+	 * Execute the Api operation.
+	 *
+	 * @return  mixed  Webservice object with information on success, boolean false on failure.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 * @throws  \Exception
+	 */
+	public function execute()
+	{
+		$options	= $this->getOptions();
+
+		$method		= $options->get('method', 'GET');
+		$task		= $options->get('task', '');
+		$format		= $options->get('format', '');
+		$clientName	= $options->get('webserviceClient');
+		$version	= $options->get('webserviceVersion');
+		$resourceName = $options->get('optionName');
+
+		// Map HTTP methods to service operations.
+		switch (strtolower($method))
+		{
+			case 'get':
+				$operation = !empty($task) ? 'task' : 'read';
+				break;
+
+			case 'put':
+				$operation = 'update';
+				break;
+
+			case 'post':
+				$operation = !empty($task) ? 'task' : 'create';
+				break;
+
+			case 'patch':
+				$operation = 'patch';
+				break;
+
+			case 'delete':
+				$operation = 'delete';
+				break;
+		}
+
+		$this->setOption('operation', $operation);
+
+		// If task is pointing to some other operation like apply, update or delete.
+//		if (!empty($task) && !empty($this->configuration->operations->task->{$task}['useOperation']))
```

Comment:
Be careful with tasks: it can lead to an RPC-style interface. You can use tasks/methods/operations in an overloaded POST, but then those tasks should be limited to the uniform interface, mostly limited to CRUD. In case of the necessity of needing a GET operation while using a POST the task should be 'read' and is handled in the current code at line 98: $operation = !empty($task) ? 'task' : 'create'; . No additional tasks are needed when using an uniform interface.